### PR TITLE
[Storage] DataLake download APIs feature parity MERGE

### PR DESCRIPTION
**Pull request by nickliu-msft**
Continued from https://github.com/Azure/azure-sdk-for-net/pull/45498

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
Continued from https://github.com/Azure/azure-sdk-for-net/pull/45498

**Comment by seanmcc-msft** [comment]
@nickliu-msft, thanks for putting this new PR together.  Looks good to me, but we are going to hold off merging until stg92base2 until the Arch Board meetings.

File: sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Net.Mime;
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs:8]: NIT - did you mean to add this?

**Review by amnguye — APPROVED** [review]
LGTM - one nit comment

**Pull request merged**